### PR TITLE
Cie 4087 security config UI

### PR DESCRIPTION
**Pull request by sharath-jk**

### CODE REVIEW

File: content/edge/configuration-bundle/configuring-security.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+   You can either import a JSON file to prefill the values or enter the values manually.  
+
+5. Select the **SELinux mode** to *permissive* or *enforcing*.
+
+   - **permissive** - In the permissive mode, the system acts as if SELinux is enforcing the loaded security policy, including labeling objects and emitting access denial entries in the logs.
+   - **enforcing** - In enforcing mode SELinux operates normally, enforcing the loaded security policy on the entire system.
+
+6. **SSH enabled** - Use the toggle button to enable or disable the SSH functionality of the Edge appliance.
+
+7. **Login sessions inactivity timeout (seconds)** - Specify the idle duration before a user session is terminated.
```
Comment (BeateRixen) [inline content/edge/configuration-bundle/configuring-security.md:33]: @sharath-jk A lot of "specify" here and below. Usually we use the format: "Specify the following parameters (or properties) followed by a table or a list.

Comment (sharath-jk) [reply content/edge/configuration-bundle/configuring-security.md:33]: Thank you, Beate. I have added the tables now.

File: content/edge/configuration-bundle/configuring-security.md [hunk 2]
```diff
@@ -21,39 +17,26 @@ In the operating system configuration, configure the SSH connectivity and the id
...
-6. Click **Save**.
-
-### Configuring the Kubernetes audit policy
-
-1. Log in to the {{< management-tenant >}} using the Edge administrator credentials created during the installation.
+   You can either import a JSON file to prefill the values or enter the values manually.
 
-   - Username: management/<*Edge admin username*>
-   - Password: password provided during the installation
+   Click **Download configuration** to download a sample JSON syntax for the current configuration. You can use the same JSON file in the POST operation using the REST API.
```
Comment (bbyreddy) [inline content/edge/configuration-bundle/configuring-security.md:22]: - [ ] Download Configuration button is in the view Security Configuration screen and not in the Edge Configuration screen.

**Review by bbyreddy — COMMENTED** [review]
- [x] @sharath-jk , this looks great. Please add the structure to the description of the parameters and the same can be replicated in the OpenAPI spec as well.
For example:
1. Basic description of the parameter
2. Its default/min/max values where applicable
3. Effects of changing this parameter. Like enabling auditing will increase the disk usage, changing the login session inactivity time out, by disabling the SSH you will not be able to SSH into the Edge appliance, neither locally not remotely etc..

**Review by BeateRixen — APPROVED** [review]

File: content/edge/configuration-bundle/configuring-security.md (hunk 2 above)
Comment (sharath-jk) [reply content/edge/configuration-bundle/configuring-security.md]: @bbyreddy - I have updated the steps.

**Comment by sharath-jk** [comment]
@bbyreddy - you can also review the output if it's convenient:
http://vmbandoc.eur.ad.sag/c8y/edge/configuration/#configuring-security

File: content/edge/configuration-bundle/configuring-security.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  ||SSH|**SSH enabled**|Enables or disables the SSH functionality of the Edge appliance. By default, the SSH functionality is enabled.<br><br>Changing the value to false disables the SSH functionality. When disabled, you will not be able to SSH into the Edge appliance, locally or remotely.<br><br>Default value: True
+  ||Session inactivity|**Login sessions inactivity timeout (seconds)**|The idle duration before a user session is terminated. When configured, the user session terminates after the specified idle duration.<br><br>Default value: 600 seconds<br>Minimum value: 0<br>Maximum value: 4294967295
+  ||Audit logging|**Audit logging enabled**|Enables or disables audit logging. By default, audit logging is disabled. A value of true enables audit logging. Enabling audit logging decreases the disk space.<br><br>Default value: False
```
Comment (bbyreddy) [inline content/edge/configuration-bundle/configuring-security.md:30]: - [ ] Enabling the audit logging "INCREASES" not decreases the disk space requirement. In fact this doesn't increase the disk space, rather increases disk space requirement or usage as a lot of data/logs are written to the disk.
- [ ] Once enabled, you can't disable the audit logging configuration. This is important to be noted. 

Check with Neeraj as he also noted this in the UI and the text there should be in sync with that in the doc.

File: content/edge/configuration-bundle/configuring-security.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  |||**Server**|The IP address or the hostname of the destination server.
+  |||**Port**|The port on the destination server.
+  |||**Protocol**|The protocol (TCP or UDP) used to transfer the logs.
+  ||Login banner|**Login banner**|The login banner for your Edge appliance.
+  |Kubernetes|Audit policy||Audit policy defines rules about what events should be recorded and what data they should include.
+  |||**Level**|The audit level of the event controls what data is recorded. When configured, the data is logged to the specified audit level. Logging more data decreases the disk space. <br><br>**None** - do not log events.<br>**Metadata** - log request metadata (requesting user, timestamp, resource, verb, etc.) but not request or response body.<br>**Request** - log event metadata and request body but not response body. This does not apply for non-resource requests.<br>**RequestResponse** - log event metadata, request and response bodies. This does not apply for non-resource requests.<br><br>Default value: None
```
Comment (bbyreddy) [inline content/edge/configuration-bundle/configuring-security.md:40]: - [ ] Logging more data "INCREASES" not decreases the disk usage. As noted in my earlier comment, the disk usage increases are more data is logged, it doesn't increase the disk space :)

File: content/edge/configuration-bundle/configuring-security.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  ||SSH|**SSH enabled**|Enables or disables the SSH functionality of the Edge appliance. By default, the SSH functionality is enabled.<br><br>Changing the value to false disables the SSH functionality. When disabled, you will not be able to SSH into the Edge appliance, locally or remotely.<br><br>Default value: True
+  ||Session inactivity|**Login sessions inactivity timeout (seconds)**|The idle duration before a user session is terminated. When configured, the user session terminates after the specified idle duration.<br><br>Default value: 600 seconds<br>Minimum value: 0<br>Maximum value: 4294967295
```
Comment (bbyreddy) [inline content/edge/configuration-bundle/configuring-security.md:29]: - [ ] Check with Neeraj regarding the UI validation he added to check the max value. If it is such a huge number, don't mention it in the doc.

File: content/edge/configuration-bundle/configuring-security.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  |||**Maximum age (days)**|The maximum number of days to retain the old audit log files. A high value has more impact on the disk space.<br><br>Default value: 30 days<br>Minimum value: 0<br>Maximum value: 2147483647
+  |||**Maximum numbers of log files to retain**|The maximum number of audit log files to retain. Setting a value of 0 indicates that there is no restriction on the number of files to retain. A high value has more impact on the disk space.<br><br>Default value: 10 files<br>Minimum value: 0<br>Maximum value: 2147483647
+  |||**Maximum size (megabytes)**|The maximum size in megabytes of the audit log file before it gets rotated. A high value has more impact on the disk space.<br><br>Default value: 100 MB<br>Minimum value: 0<br>Maximum value: 2147483647
```
Comment (bbyreddy) [inline content/edge/configuration-bundle/configuring-security.md:43]: - [ ] Check with Neeraj regarding the UI validation he added to check the max value. If it is such a huge number, don't mention it in the doc.

**Review by bbyreddy — COMMENTED** [review]
@sharath-jk , please find my comments.

File: content/edge/configuration-bundle/configuring-security.md (hunk 4 above)
Comment (sharath-jk) [reply content/edge/configuration-bundle/configuring-security.md:40]: @bbyreddy - I have mentioned it as "decreases the disk space".

Do you want me to change it to "decreases the available disk space" or "increases the disk space usage"?

File: content/edge/configuration-bundle/configuring-security.md (hunk 3 above)
Comment (sharath-jk) [reply content/edge/configuration-bundle/configuring-security.md:30]: @bbyreddy - I have mentioned it as "decreases the disk space".

Do you want me to change it to "decreases the available disk space" or "increases the disk space usage"?

File: content/edge/configuration-bundle/configuring-security.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+4. Click **Download configuration** to download a sample JSON syntax for the current configuration. You can use the same JSON file in the POST operation using the REST API.
+
+5. Click **Edit configuration** to edit the configuration. 
+
+   You can either import a JSON file to prefill the values or enter the values manually.
+
+6. Specify the following parameters:
+  |<div style="width:100px">Configuration</div>|Category|Parameter|Description
+  |:---|:---|:---|:---
+  |Operating system|SELinux|**SELinux mode**|SELinux secures the operating system from unprivileged actions of confined users. When enabled, SELinux has two modes: permissive and enforcing.<br><br>**permissive**: In the permissive mode, SELinux policy is not enforced. The system remains operational and SELinux does not deny any operations but only logs AVC messages, which can be then used for troubleshooting, debugging, and SELinux policy improvements. Each AVC is logged only once in this case.<br><br>**enforcing**: In the enforcing mode, it enforces the SELinux policy and denies access based on SELinux policy rules.<br><br>Default value: permissive
```
[the remark this answers is not in the document]
Comment (sharath-jk) [reply content/edge/configuration-bundle/configuring-security.md:27]: I've updated the text after checking with Neeraj.

File: content/edge/configuration-bundle/configuring-security.md (hunk 3 above)
Comment (sharath-jk) [reply content/edge/configuration-bundle/configuring-security.md:30]: @bbyreddy - I have added the text "Once enabled, you cannot disable the audit logging configuration."

Also, updated the text to "Enabling audit logging increases the consumption of disk space."

File: content/edge/configuration-bundle/configuring-security.md (hunk 4 above)
Comment (sharath-jk) [reply content/edge/configuration-bundle/configuring-security.md:40]: @bbyreddy - Updated the text "Logging more data increases the consumption of disk space."

File: content/edge/configuration-bundle/configuring-security.md (hunk 5 above)
Comment (sharath-jk) [reply content/edge/configuration-bundle/configuring-security.md:29]: @bbyreddy - Neeraj confirmed that the Maximum value is a huge number as mentioned in the doc. So removing the maximum value.

File: content/edge/configuration-bundle/configuring-security.md (hunk 6 above)
Comment (sharath-jk) [reply content/edge/configuration-bundle/configuring-security.md:43]: @bbyreddy - Neeraj confirmed that the Maximum value is a huge number as mentioned in the doc. So removing the maximum value.

**Review by bbyreddy — APPROVED** [review]
@sharath-jk , I hope the same changes have been replicated in the OpanAPI spec as well. If not, please update them.
Thanks!

**Comment by sharath-jk** [comment]
@bbyreddy - Yes, same has been replicated in the Open API. Here is the PR for it:
https://github.softwareag.com/IOTA/cumulocity-edge-openapi/pull/17

**Comment by rssag** [comment]
hello!
to which releases does this apply? would you like me to cherry-pick it somewhere? @sharath-jk

**Comment by sharath-jk** [comment]
@rssag - This applies to 10.15 release. I have already cherry picked this to release/r10.15.0 branch.

Here is the commit:
https://github.com/SoftwareAG/c8y-docs/commit/0622b973547bf642ab8c5ca8e3ef6c1eb88c5702

**Comment by rssag** [comment]
> @rssag - This applies to 10.15 release. I have already cherry picked this to release/r10.15.0 branch.
> 
> Here is the commit: [0622b97](https://github.com/SoftwareAG/c8y-docs/commit/0622b973547bf642ab8c5ca8e3ef6c1eb88c5702)

ah, thank you :)

**Pull request merged**